### PR TITLE
prevent invalid container name

### LINKED ISSUE #102: directories don't work for blobs that start with '/'
silly corner case.

### PR DESCRIPTION
**Pull request by cicorias**
Closes #102

### CODE REVIEW

**Comment by felixrieseberg** [comment]
Hey, thanks for the PR! We could run with the helper to display an error message in the template. In that case, could you remove the changes made to the controller? Otherwise, LGTM.

**Comment by felixrieseberg** [comment]
Actually, I didn't think straight - I take it all back. We should _not_ use a helper, but instead move the whole thing into the controller. Kinda like this, on the welcome controller:

```
containerNameIsValid: function (name) {
    let regex = /^(([a-z\d]((-(?=[a-z\d]))|([a-z\d])){2,62})|(\$root)|(\$logs))$/;
    return regex.test(name);
}
```

The reason: Helpers are run in the rendering loop by HTMLBars/Handlebars. This is great for templating logic, but we're dealing more with validation logic here - and should do that in the controller (or on a component), but not in the render phase.

**Comment by cicorias** [comment]
OK, let me revise.

**Comment by cicorias** [comment]
@felixrieseberg  revised LGTM

File: app/controllers/explorer.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import Ember from 'ember';
 import Notification from '../models/notification';
 import stringResources from '../utils/string-resources';
+import containerUtil from '../utils/container';
```
Comment (sedouard) [inline app/controllers/explorer.js]: This doesn't seem to be right. I don't think this exists.

File: app/utils/string-resources.js [hunk 2]
```diff
@@ -46,5 +46,8 @@ export default {
     },
     storageRejectedAccountKey: function () {
         return 'The connection succeeded, but the Azure rejected the account key. Please check it and try again.';
-    }
+    },
+		containerValidNames: function () {
```
Comment (sedouard) [inline app/utils/string-resources.js]: `string-resources` is meant specifically for all the messages we display via code. (so in the future we can localize stuff if we need to).

Comment (sedouard) [inline app/utils/string-resources.js]: I don't see the point of breaking this out into a util module, it won't really be reused anywhere else.

File: app/controllers/explorer.js [hunk 3]
```diff
@@ -549,7 +550,14 @@ export default Ember.Controller.extend({
             if (!name){
                 return;
             }
-            name = name.toLowerCase();
+
+						// Todo - something to signal user that the container name was invalid
+						name = name.toLowerCase();
+
+						if (!containerUtil.containerNameIsValid(name)){
```
Comment (sedouard) [inline app/controllers/explorer.js]: How about we just put the regex validation, right in here?

Comment (cicorias) [inline app/controllers/explorer.js]: sure

**Comment by cicorias** [comment]
and collapse a bit

**Comment by sedouard** [comment]
@cicorias  By the way this will close https://github.com/azure-storage/deco/issues/297. #102 referrs to blob names, not container names.

If you place your regex check within `addContainerData, along with my PR to use `createContainer` instead  of `createContinerIfNotExists` this would close #297 and we'd be good to go.

If you want to get fancy and notify the user of the validation error (instead of `console.warning`) you can do:

```js
this.get('notifications').addNotification(Notification.create(
                {
                    type: 'AddContainer',
                    // add the function below to stringResources module
                    text: stringResources.addContainerInvalidMessage(name),
                    // indicates it errored out
                    progress: -1
                })
);
```

This would show the error in the notifications panel on the bottom. I can do this if you don't feel like it either.

**Comment by cicorias** [comment]
let me restart this one... way out of sync

**Pull request closed without merging**